### PR TITLE
Pin `pint<0.22` to fix euphonic error

### LINKED ISSUE #35662: AbinsLoadPhonopyTest is failing due to pint update
**Describe the bug**
The `euphonic` package we use has a dependency to `pint`. A more recent version of `pint` has been uploaded to conda (v0.22 noarch), and this update is a breaking change to the `euphonic` package we rely on.

This means the `AbinsLoadPhonopyTest` has started failing, and will begin to fail on all unrelated PRs being merged to `main` or `release-next`. See the failure here:

https://builds.mantidproject.org/job/pull_requests-conda-osx/2767/testReport/projectroot.scripts.test/Abins/python_scripts_AbinsLoadPhonopyTest_AbinsLoadPhonopyTest/

**To Reproduce**
Read the test output here:
https://builds.mantidproject.org/job/pull_requests-conda-osx/2767/testReport/projectroot.scripts.test/Abins/python_scripts_AbinsLoadPhonopyTest_AbinsLoadPhonopyTest/

**Platform/Version (please complete the following information):**
All platforms

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work**
This PR pins pint to be less than v0.22 in order to fix a breaking change causing an error in euphonic.

**To test:**
Make sure `AbinsLoadPhonopyTest` passes in our CI

Fixes #35662 

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by robertapplin** [comment]
This is a temporary fix so that are builds can start passing again. We need to wait for the issue in the `pint-feedstock` to be resolved before this can be unpinned in our repository https://github.com/conda-forge/pint-feedstock/issues/61

File: conda/recipes/mantid/meta.yaml [hunk 1]
```diff
@@ -65,6 +65,7 @@ requirements:
     - scipy
     - openssl
     - euphonic=0.6.*
+    - pint<0.22  # version 0.22 introduces a breaking change to euphonic 0.6.*
```
Comment (MialLewis) [inline conda/recipes/mantid/meta.yaml:68]: Shall we move the `euphonic` and `pint` pins to the `conda_build_config.yaml` folder for consistency?

**Review by thomashampson — CHANGES_REQUESTED** [review]
If we use `!=0.22` then we won't have to unpin it in the future. See here for example:
https://github.com/mantidproject/mantid/pull/35510/files

File: conda/recipes/mantid/meta.yaml (hunk 1 above)
Comment (thomashampson) [reply conda/recipes/mantid/meta.yaml:68]: yes that would be good

Comment (ajjackson) [reply conda/recipes/mantid/meta.yaml:68]: Is this file now the place to make dependency changes, or do we still need to also make a PR against the conda-forge feedstock?

Comment (robertapplin) [reply conda/recipes/mantid/meta.yaml:68]: By the conda-forge feedstock, do you mean the old `conda-recipes` repo? I believe this repo has now been moved into the mantid repo @thomashampson ?

**Review by thomashampson — APPROVED** [review]
Looking good

File: conda/recipes/mantid/meta.yaml (hunk 1 above)
Comment (ajjackson) [reply conda/recipes/mantid/meta.yaml]: Yes that's the one, this is where the dependency was added initially https://github.com/mantidproject/conda-recipes/pull/74

**Pull request merged**